src/x86_64/avx2.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2024 the Fearless_SIMD Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Support for the Avx2 level.
+
+use crate::{
+    f32x4, f32x8,
+    impl_macros::{impl_op, impl_simd_from_into},
+    mask32x4, mask32x8,
+    seal::Seal,
+    Simd, SimdFrom, SimdInto,
+};
+use core::arch::x86_64::*;
+
+use super::Level;
+
+/// The SIMD token for the "avx2" level.
+///
+/// This is short for the "x86-64-v3 microarchitecture level". In this level, the
+/// following target_features are enabled: "avx2", "bmi2", "f16c", "fma", "lzcnt".
+
+/// The SIMD token for the "avx2" level.
+///
+/// This is short for the "x86-64-v3 microarchitecture level". In this level, the
+/// following target_features are enabled: "avx2", "bmi2", "f16c", "fma", "lzcnt".
```

Comment:
Is authoritative documentation available somewhere of CPU feature hierarchies?

Reply:
That's a *good* question, and when you look at other implementations you see immense confusion and potential bugs that will impact fringe x86 clones (I linked at least one such bug from my [irlo thread](https://internals.rust-lang.org/t/better-codegen-for-cpu-feature-detection/22083)).

I'm generally going off [Wikipedia](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels) for the v3 etc microarchitecture levels. In terms of the graph of implied features, my main source is [rustc](https://github.com/rust-lang/rust/blob/52890e82153cd8716d97a96f47fb6ac99dec65be/compiler/rustc_target/src/target_features.rs#L207).

Probably this should be captured in a wiki; there's a ton of lore and I'm not aware of any great authoritative sources.